nbconvert/nbconvertapp.py

```diff
@@ -323,16 +330,21 @@ def init_single_notebook_resources(self, notebook_filename):
 
         return resources
 
-    def export_single_notebook(self, notebook_filename, resources):
+    def export_single_notebook(self, notebook_filename, resources, input_buffer=None):
         """Step 2: Export the notebook
 
         Exports the notebook to a particular format according to the specified
         exporter. This function returns the output and (possibly modified)
         resources from the exporter.
 
+        notebook_filename: a filename
+        input_buffer: a containing notebook buffer, if not None notebook_filename is ignored
```

Comment:
This description wasn't completely clear to me. The convention I'm used to in Python is to call these 'file like objects'. Annoyingly, there's a lot of 'file like objects' with slightly different properties, so it's a good idea to clarify, e.g.

> a readable file like object returning unicode